src/plugins/pagination/pager-target/pager-target.component.ts

```diff
@@ -0,0 +1,39 @@
+import { Component, Optional, OnInit } from '@angular/core';
+import { PluginComponent } from '../../plugin.component';
+import { RootService } from 'ng2-qgrid/infrastructure/component/root.service';
+
+@Component({
+	selector: 'q-grid-pager-target',
+	templateUrl: './pager-target.component.html'
+})
+export class PagerTargetComponent extends PluginComponent implements OnInit {
+	constructor(@Optional() root: RootService) {
+		super(root);
+	}
+
+	private value: number;
+	private target: any;
+
+	ngOnInit() {
+		this.target = this;
+		this.context = {$implicit: this.target};
+	}
+
+	setPage(page: number) {
+		const maxAllowed = this.total();
+
+		if (page > 0 && page <= maxAllowed) {
```

Comment:
when this code is triggered? i can see max/min constraints in the input?

Reply:
its here https://github.com/qgrid/ng2/pull/164/files/3ff88e2fa95483bf3ea252e53618228470b71055#diff-af5de63a7ed3c19630eeae22c1b7dfc2

Reply:
yes i know, but when $target.value can be not valid if you set min max?

Reply:
We still have this validation why?